fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LucenePartitioner.java

```diff
@@ -1092,43 +1106,108 @@ private CompletableFuture<Integer> moveDocsFromPartition(@Nonnull final LuceneRe
-                    });
+            for (FDBIndexableRecord<Message> record : records) {
+                LuceneDocumentFromRecord.getRecordFields(state.index.getRootExpression(), record)
+                        .entrySet().forEach(entry -> {
+                            indexMaintainer.writeDocument(record, entry, destinationPartitionId);
+                            // TODO could update the partition once
+                            addToAndSavePartitionMetadata(record, groupingKey, destinationPartitionId);
+                        });
+            }
+            if (LOGGER.isDebugEnabled()) {
+                long updateNanos = System.nanoTime();
+                final KeyValueLogMessage logMessage = repartitionLogMessage("Repartitioned records", groupingKey, records.size(), partitionInfo);
+                logMessage.addKeyAndValue("totalMicros", TimeUnit.NANOSECONDS.toMicros(updateNanos - timings.startNanos));
+                logMessage.addKeyAndValue("initializationMicros", TimeUnit.NANOSECONDS.toMicros(timings.initializationNanos - timings.startNanos));
+                logMessage.addKeyAndValue("searchMicros", TimeUnit.NANOSECONDS.toMicros(timings.searchNanos - timings.initializationNanos));
+                logMessage.addKeyAndValue("clearInfoMicros", TimeUnit.NANOSECONDS.toMicros(timings.clearInfoNanos - timings.searchNanos));
+                if (timings.emptyingNanos > 0) {
+                    logMessage.addKeyAndValue("emptyingMicros", TimeUnit.NANOSECONDS.toMicros(timings.emptyingNanos - timings.clearInfoNanos));
+                }
+                if (timings.deleteNanos > 0) {
+                    logMessage.addKeyAndValue("deleteMicros", TimeUnit.NANOSECONDS.toMicros(timings.deleteNanos - timings.clearInfoNanos));
+                }
+                if (timings.metadataUpdateNanos > 0) {
+                    logMessage.addKeyAndValue("metadataUpdateMicros", TimeUnit.NANOSECONDS.toMicros(timings.metadataUpdateNanos - timings.deleteNanos));
+                }
+                if (timings.createPartitionNanos > 0) {
+                    logMessage.addKeyAndValue("createPartitionMicros", TimeUnit.NANOSECONDS.toMicros(timings.createPartitionNanos - endCleanupNanos));
+                }
+                logMessage.addKeyAndValue("updateMicros", TimeUnit.NANOSECONDS.toMicros(updateNanos - updateStart));
+                if (timerSnapshot != null && state.context.getTimer() != null) {
+                    logMessage.addKeysAndValues(
+                            StoreTimer.getDifference(state.context.getTimer(), timerSnapshot)
+                                    .getKeysAndValues());
+                }
+                LOGGER.debug(logMessage.toString());
+            }
+            return records.size();
         });
     }
 
+    @Nonnull
+    private CompletableFuture<Integer> removeEmptyPartition(@Nonnull final LuceneRepartitionPlanner.RepartitioningContext repartitioningContext) {
+        // sanity check
```

Comment:
Should these two sanity checks throw internal errors instead of logging a debug log?